synapse/storage/client_ips.py

```diff
@@ -122,17 +127,20 @@ def get_last_seen(txn):
                 LIMIT 1
                 OFFSET ?
                 """,
-                (last_seen_progress, batch_size)
+                (begin_last_seen, batch_size)
             )
-            results = txn.fetchone()
-            return results
-
-        # Get a last seen that's sufficiently far away enough from the last one
-        last_seen = yield self.runInteraction(
+            row = txn.fetchone()
+            if row:
+                return row[0]
+            else:
+                return None
```

Comment:
This is just moving the `row[0]` from the if statement below into the function, as it feels cleaner